tests/contrib/asgi/test_asgi.py

```diff
@@ -0,0 +1,260 @@
+import asyncio
+from functools import partial
+import random
+import sys
+
+import httpx
+import pytest
+from asgiref.testing import ApplicationCommunicator
+from ddtrace.contrib.asgi import TraceMiddleware
+from ddtrace.propagation import http as http_propagation
+from tests.base import BaseTestCase
+from tests.test_tracer import get_dummy_tracer
+
+
+@pytest.fixture(
+    params=[
+        {},
+        {"server": None},
+        {"http_version": "1.0", "asgi": {}},
+        {"http_version": "1.0", "asgi": {"version": "3.2"}},
+        {"http_version": "1.0", "asgi": {"spec_version": "2.1",}},
+        {"http_version": "1.0", "asgi": {"version": "3.2", "spec_version": "2.1",}},
+    ]
+)
+def scope(request):
+    s = {
+        "client": ("127.0.0.1", 32767),
+        "headers": [],
+        "method": "GET",
+        "path": "/",
+        "query_string": b"",
+        "scheme": "http",
+        "server": ("127.0.0.1", 80),
```

Comment:
Would be good to test with different ports since we have logic around that

Reply:
👍